### PR TITLE
Enhancement/add lifecycle notebook sm

### PR DESCRIPTION
**Pull request by StuartLox**
Enhancement #2999

Changes proposed in this pull request:

* Adding lifecycle configuration name as an optional param for Create Notebook Instance
* Update sagemaker documentation to include lifecycle configuration name.

### CODE REVIEW

**Comment by StuartLox** [comment]
Just wondering if there's anything else that needs to be done on my end to get this approved?

**Comment by bcatubig** [comment]
@StuartLox Looks good to me. I would add a test case as well to this pr. You probably want to remove the vendor files as well

File: go.mod [hunk 1]
```diff
@@ -28,11 +28,11 @@ require (
 	github.com/hashicorp/go-version v1.0.0
 	github.com/hashicorp/hcl v0.0.0-20170504190234-a4b07c25de5f // indirect
 	github.com/hashicorp/hcl2 v0.0.0-20171003232734-44bad6dbf549 // indirect
-	github.com/hashicorp/hil v0.0.0-20170512213305-fac2259da677 // indirect
+	github.com/hashicorp/hil v0.0.0-20190212132231-97b3a9cdfa93 // indirect
```
Comment (bflad) [inline go.mod]: I'd recommend removing any `go.mod` and `go.sum` changes to this pull request as they should be unrelated to this change and likely cause merge conflicts (as they are currently).

**Comment by bflad** [comment]
This has been released in [version 2.5.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#250-april-05-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**